types/stylelint/index.d.ts

```diff
@@ -1,15 +1,28 @@
 declare module 'stylelint' {
 	import { Comment, Result, ResultMessage, Root, Syntax, WarningOptions, Warning } from 'postcss';
 
+	export type Severity = 'warning' | 'error';
+
 	export type StylelintConfigExtends = string | Array<string>;
 	export type StylelintConfigPlugins = string | Array<string>;
 	export type StylelintConfigProcessor = string | [string, Object];
 	export type StylelintConfigProcessors = string | Array<StylelintConfigProcessor>;
 	export type StylelintConfigIgnoreFiles = string | Array<string>;
-	export type StylelintConfigRuleSettings = any | [any, Object];
+	export type StylelintConfigRuleSettings<T, O extends Object> =
+		| null
+		| undefined
+		| NonNullable<T>
+		| [NonNullable<T>]
+		| [NonNullable<T>, O];
 	export type StylelintConfigRules = {
-		[ruleName: string]: StylelintConfigRuleSettings;
+		[ruleName: string]: StylelintConfigRuleSettings<any, Object>;
+	};
+
+	export type DisableOptions = {
+		except?: Array<string | RegExp>;
```

Comment:
Would it make sense to add a test where this is a regex? Or is this sufficiently covered by existing tests for `optionsMatches`?

Reply:
It looks like other rules that allow RegExps don't explicitly test them, so I think it's probably covered.